modal/cli/dict.py

```diff
@@ -0,0 +1,123 @@
+    """Create a named Dict object."""
+    d = _Dict.from_name(name, environment_name=env, create_if_missing=True)
+    client = await _Client.from_env()
+    resolver = Resolver(client=client)
+    await resolver.load(d)
+
+
+@dict_cli.command(name="list")
+@synchronizer.create_blocking
+async def list(*, json: bool = False, env: Optional[str] = ENV_OPTION):
+    """List all named Dict objects."""
+    env = ensure_env(env)
+    client = await _Client.from_env()
+    request = api_pb2.DictListRequest(environment_name=env)
+    response = await retry_transient_errors(client.stub.DictList, request)
+
+    def format_timestamp(t: float) -> str:
+        return datetime.strftime(datetime.fromtimestamp(t), "%Y-%m-%d %H:%M") + " UTC"
+
+    rows = [(d.name, format_timestamp(d.created_at)) for d in response.dicts]
+    display_table(["Name", "Created at"], rows, json)
+
+
+@dict_cli.command("clear")
+@synchronizer.create_blocking
+async def clear(name: str, *, env: Optional[str] = ENV_OPTION):
+    """Clear the contents Dict by deleting all of its data."""
+    d = await _Dict.lookup(name, environment_name=env)
+    await d.clear()
+
+
+@dict_cli.command(name="delete")
+@synchronizer.create_blocking
+async def delete(name: str, *, env: Optional[str] = ENV_OPTION):
+    """Delete a named Dict object and all of its data."""
+    client = await _Client.from_env()
+    lookup_request = api_pb2.AppGetByDeploymentNameRequest(
+        name=name,
+        environment_name=ensure_env(env),
+        namespace=api_pb2.DEPLOYMENT_NAMESPACE_WORKSPACE,
+    )
+    resp = await client.stub.AppGetByDeploymentName(lookup_request)
+    stop_req = api_pb2.AppStopRequest(app_id=resp.app_id, source=api_pb2.APP_STOP_SOURCE_CLI)
+    await client.stub.AppStop(stop_req)
+
+
+@dict_cli.command(name="get")
+@synchronizer.create_blocking
+async def get(name: str, key: str, *, env: Optional[str] = ENV_OPTION):
+    """Print the value for a specific key.
+
+    Note: When using the CLI, keys are always interpreted as having a string type.
+    """
+    d = await _Dict.lookup(name, environment_name=env)
+    console = Console()
+    val = await d.get(key)
+    console.print(val)
+
+
+@dict_cli.command(name="show")
+@synchronizer.create_blocking
+async def show(
+    name: str,
+    n: int = Argument(default=None, help="Retrieve and show no more than this many entries"),
+    *,
+    json: bool = False,
+    env: Optional[str] = ENV_OPTION,
+):
+    """Print the contents of a Dict.
+
+    Note: By default, this command will retrieve the complete Dict contents.
+    Using the N argument is recommended for quickly inspecting a small number of items.
+    """
+    d = await _Dict.lookup(name, environment_name=env)
+
+    try:
+        i, items = 0, []
+        async for item in d.items():
+            i += 1
+            items.append(item)
+            if n is not None and i >= n:
+                break
+    except ModuleNotFoundError as exc:
```

Comment:
maybe this except block should be in the `items()` method?

Reply:
sounds good! We actually have some code here that handles a similar issue: https://github.com/modal-labs/modal-client/blob/main/modal/_utils/function_utils.py#L450